src/gui/mainwindow.cpp

```diff
@@ -1475,6 +1482,33 @@ void MainWindow::showStatusBar(bool show)
     }
 }
 
+void MainWindow::showFiltersSidebar(const bool show)
+{
+    Preferences *const pref = Preferences::instance();
+
+    if (show)
+    {
+        const int width = pref->getFiltersSidebarWidth();
+        m_transferListFiltersWidget = new TransferListFiltersWidget(m_splitter, m_transferListWidget, isDownloadTrackerFavicon());
+        m_splitter->insertWidget(0, m_transferListFiltersWidget);
+        m_splitter->setCollapsible(0, true);
+        m_splitter->setSizes({width, (m_splitter->width() - width)});
+
+        connect(BitTorrent::Session::instance(), &BitTorrent::Session::trackersAdded, m_transferListFiltersWidget, &TransferListFiltersWidget::addTrackers);
+        connect(BitTorrent::Session::instance(), &BitTorrent::Session::trackersRemoved, m_transferListFiltersWidget, &TransferListFiltersWidget::removeTrackers);
+        connect(BitTorrent::Session::instance(), &BitTorrent::Session::trackerlessStateChanged, m_transferListFiltersWidget, &TransferListFiltersWidget::changeTrackerless);
+        connect(BitTorrent::Session::instance(), &BitTorrent::Session::trackerEntriesUpdated, m_transferListFiltersWidget, &TransferListFiltersWidget::trackerEntriesUpdated);
+    }
+    else
+    {
+        Q_ASSERT(m_splitter->widget(0) == m_transferListFiltersWidget);
```

Comment:
I would add counterpart assertion in `if` branch or (maybe better) make the method "safe", e.g.:
```c++
if (show && !m_transferListFiltersWidget)
    //...
else if (!show && m_transferListFiltersWidget)
    //...
```